internal/kibana/saved_object/read.go

```diff
@@ -0,0 +1,45 @@
+package saved_object
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func (r *Resource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var model ksoModelV0
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	kibanaClient, err := r.client.GetKibanaClient()
+	if err != nil {
+		resp.Diagnostics.AddError("unable to get kibana client", err.Error())
+		return
+	}
+
+	result, err := kibanaClient.KibanaSavedObject.Get(model.Type.ValueString(), model.ID.ValueString(), model.SpaceID.ValueString())
```

Comment:
Reading saved objects has been [deprecated](https://github.com/elastic/terraform-provider-elasticstack/issues/637) since 8.7. It'll be completely removed shortly, we could keep this for versions which support the `Get`, but IMO the UX where drift detection is silently dropped when upgrading the underlying Kibana version is pretty bad.

I think the `Read` should be basically a no-op here, simply maintaining the current state.

Reply:
I was not aware of the deprecation of the API. I guess this PR doesn't make such sense then :(

Crossplane relies on the Read to be functional to work. That's the whole point of this PR. Without this, the reconciliation loop can't work.

There's a new API in tech preview to manage dashboards, I think it makes sense to refactor this code to use this new API.